File: C++/1614-maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) {
      int result = 0, depth = 0;
      for(char& i:s){
        if(i == '(')
          result = max(result, ++depth);
        if (i == ')')
          depth--;
      }
      return result;
    }
};

File: Java/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
      int res = 0, depth = 0;
      for(int i = 0; i < s.length(); i++){
        if(s.charAt(i) == '('){
          res = Math.max(res, ++depth);
        }
        if(s.charAt(i)==')')
          depth--;
      }
      return res;
    }
}
